Makefile

```diff
@@ -2,10 +2,19 @@
 all: buildroot/Makefile buildroot/.config
 	make -C buildroot
 
+clean:
+	BR2_EXTERNAL=.. make -C buildroot clean
+
 buildroot/Makefile:
 	git submodule update --init buildroot
 
 buildroot/.config:
 	BR2_EXTERNAL=.. make -C buildroot fischertechnik_TXT_defconfig
 
-
+release:
+	if [ ! -f $(imagedir)/rootfs.img ]; then make; fi
```

Comment:
Why not simply depend on 'all' for the release target and skip the rucursive make call from a called shell?

Especially since $(imagedir) isn't defined yet when this line is executed, and consequently 'make all' is always triggered anyway if you don't happen to have a "rootfs.img" lying around in your hosts root directory :grin:

Reply:
Depending on all costs extra time. This way the script only builds if needed.

Reply:
Only if you already did a "make" just before the release, and did not change anything since. And that is a very special edge case (my guess: It's really only relevant while you are testing the "release" target :smile:) . 

Normally, you *want* to do a build (and ideally even a clean build) immediately before releasing. And doing a relase is a pretty rare activity anyway (at least compared to normal builds), so wasting a few seconds waiting fo buildroot to figure out that there's nothing to rebuild in the rare cases that you already manually triggered that build is not much of an issue in my opinion...